Sources/AsyncHTTPClient/NIOTransportServices/TLSConfiguration.swift

```diff
@@ -102,6 +97,11 @@
                 preconditionFailure("TLSConfiguration.keyLogCallback is not supported")
             }
 
+            // the certificate chain
+            if self.certificateChain.count > 0 {
+                preconditionFailure("TLSConfiguration.certificateChain is not supported")
```

Comment:
is crashing appropriate here? I'd imagine that other libraries may be handing through their `TLSConfiguration`s?

Reply:
This was what it was doing before. I'm happy to throw an error instead, but that pretty much amounts to the same thing: the user of AHC has to remove the certificate chain and private key from their `TLSConfiguration`. We certainly cannot fail silently as people will be expecting their certificate chain/private key to be taken into account if we don't provide some error of some sort.

Reply:
I think crashing is fine here until we add actual support.